src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -35,6 +47,207 @@ public ProjectView()
             _versions.ItemContainerStyle = style;
         }
 
+        private string _previousFilter;
+        private string PreviousFilter
+        {
+            get
+            {
+                if (_previousFilter != null) return _previousFilter;
+                return _versions.SelectedValue.ToString();
+            }
+            set
+            {
+                _previousFilter = value;
+                OnPropertyChanged(nameof(PreviousFilter));
+            }
+        }
+
+        private bool VersionsFilter(object o)
+        {
+            string ComboboxText = _versions.Text;
+            if (string.IsNullOrEmpty(_versions.Text)) return true;
+            if (ComboboxText.Length == 0 && ComboboxText.Equals("*", StringComparison.OrdinalIgnoreCase)) return true;
+            if ((ComboboxText.StartsWith("(", StringComparison.OrdinalIgnoreCase) || ComboboxText.StartsWith("[", StringComparison.OrdinalIgnoreCase)) &&
```

Comment:
If this is the best way to detect ranges, can we extract it to a helper method somewhere?